src/pages/ServiceGraph/ServiceGraphPage.tsx

```diff
@@ -181,40 +170,12 @@ export default class ServiceGraphPage extends React.Component<ServiceGraphPagePr
 
   /** Fetch graph data */
   loadGraphDataFromBackend = (namespace?: Namespace, graphDuration?: Duration) => {
-    this.setState({ isLoading: true, isReady: false });
     namespace = namespace ? namespace : this.props.namespace;
-    const duration = graphDuration ? graphDuration.value : this.props.graphDuration.value;
-    const restParams = { duration: duration + 's' };
-    API.getGraphElements(namespace, restParams)
-      .then(response => {
-        if (!this._isMounted) {
-          console.log('ServiceGraphPage: Ignore fetch, component not mounted.');
-          return;
-        }
-        const responseData = response['data'];
-        const elements = responseData && responseData.elements ? responseData.elements : EMPTY_GRAPH_DATA;
-        const timestamp = responseData && responseData.timestamp ? responseData.timestamp : '';
-        this.setState({
-          graphData: elements,
-          graphTimestamp: timestamp,
-          summaryData: null,
-          isLoading: false
-        });
-      })
-      .catch(error => {
-        if (!this._isMounted) {
-          console.log('ServiceGraphPage: Ignore fetch error, component not mounted.');
-          return;
-        }
-        this.setState({
-          graphData: EMPTY_GRAPH_DATA,
-          graphTimestamp: new Date().toLocaleString(),
-          summaryData: null,
-          isLoading: false,
-          alertVisible: true,
-          alertDetails: { ...this.state.alertDetails, graphLoadAlert: error }
-        });
-      });
+    graphDuration = graphDuration ? graphDuration : this.props.graphDuration;
+    this.props.dispatch(ServiceGraphDataActions.fetchGraphData(namespace, graphDuration));
```

Comment:
Here's the problem I encountered and why I did this. If there is a better way, let me know.
Notice the parameters to fetchGraphData - it takes namespace and graphDuration which comes from either (a) the arguments passed to loadGraphDataFromBackend or (b) this.props. Because it needs props, I couldn't figure out how to use mapDispatchToProps to pass to connect() that can pass in things like other property values.

Reply:
Maybe i'm missing something, would this work?
```typescript
mapDispatchToProps = (dispatch: any) => ({
  fetchGraphData: (namespace, graphDuration) => dispatch(ServiceGraphDataActions.fetchGraphData(namespace, graphDuration))
});
```

Then you would replace the line in the code with 
```typescript
this.props.fetchGraphData(namespace, graphDuration);
```

Similar syntax worked for me: https://github.com/kiali/kiali-ui/pull/315/files#diff-e83846854896e8fbe8f3ef8ee4cb5f20R20

Reply:
Thanks! I didn't think to put parameters in the function defined in mapDispatchToProps. Works for me, too.